src/Numerics/DGMethods/DGModel_kernels.jl

```diff
@@ -900,74 +863,80 @@ fluxes, respectively.
+                    Vars{vars_state(balance_law, Prognostic(), FT)}(
+                        local_state_prognostic⁻,
+                    ),
+                    Vars{vars_state(balance_law, Auxiliary(), FT)}(
+                        local_state_auxiliary⁻,
+                    ),
+                    Vars{vars_state(balance_law, Prognostic(), FT)}(
+                        local_state_prognostic⁺nondiff,
+                    ),
+                    Vars{vars_state(balance_law, Auxiliary(), FT)}(
+                        local_state_auxiliary⁺nondiff,
+                    ),
+                    t,
+                    face_direction,
+                    Vars{vars_state(balance_law, Prognostic(), FT)}(
+                        local_state_prognostic_bottom1,
+                    ),
+                    Vars{vars_state(balance_law, Auxiliary(), FT)}(
+                        local_state_auxiliary_bottom1,
+                    ),
+                )
+                numerical_boundary_flux_second_order!(
+                    numerical_flux_second_order,
+                    bc,
+                    balance_law,
+                    Vars{vars_state(balance_law, Prognostic(), FT)}(local_flux),
+                    normal_vector,
+                    Vars{vars_state(balance_law, Prognostic(), FT)}(
+                        local_state_prognostic⁻,
+                    ),
+                    Vars{vars_state(balance_law, GradientFlux(), FT)}(
+                        local_state_gradient_flux⁻,
+                    ),
+                    Vars{vars_state(balance_law, Hyperdiffusive(), FT)}(
+                        local_state_hyperdiffusion⁻,
+                    ),
+                    Vars{vars_state(balance_law, Auxiliary(), FT)}(
+                        local_state_auxiliary⁻,
+                    ),
+                    Vars{vars_state(balance_law, Prognostic(), FT)}(
+                        local_state_prognostic⁺diff,
+                    ),
+                    Vars{vars_state(balance_law, GradientFlux(), FT)}(
+                        local_state_gradient_flux⁺,
+                    ),
+                    Vars{vars_state(balance_law, Hyperdiffusive(), FT)}(
+                        local_state_hyperdiffusion⁺,
+                    ),
+                    Vars{vars_state(balance_law, Auxiliary(), FT)}(
+                        local_state_auxiliary⁺diff,
+                    ),
+                    t,
+                    Vars{vars_state(balance_law, Prognostic(), FT)}(
+                        local_state_prognostic_bottom1,
+                    ),
+                    Vars{vars_state(balance_law, GradientFlux(), FT)}(
+                        local_state_gradient_flux_bottom1,
+                    ),
+                    Vars{vars_state(balance_law, Auxiliary(), FT)}(
+                        local_state_auxiliary_bottom1,
+                    ),
+                )
+            end d -> throw(BoundsError(bcs, bctag))
```

Comment:
@simonbyrne why did you have to unroll the tuple in the kernel versus in its own function like before? Curious as to why this solved the problems you were having

Reply:
the next step after this is to be able to add an extra "boundary auxiliary state", which will depend on the boundary condition. For that to work, we would need to unroll it in the kernel.

Reply:
ah got it